### PR TITLE
infoschema, executor: Add variables_info table to see sysvar details

### PR DESCRIPTION
**Pull request by joycse06**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/21446

Problem Summary:
Currently there's no easy way to find out which system variables have been changed from their default value in a TiDB Cluster which can be handy for debugging. Also currently there's no easy way to see if a variable is instance scope.

### What is changed and how it works?
What's changed:

`information_schema.variables_info` has been added. This is a view and doesn't make any changes to persistence. The data is read from `variable.GetSysVars()` method. 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
* `information_schema.variables_info` has been added which can be used to quickly check which variables has been changed from their default value.
```

### CODE REVIEW

**Review by morgo — APPROVED** [review]

**Comment by Defined2014** [comment]
@morgo Do you think move it to performance_schmea is better? Compatible with MySQL 8.0

**Review by Defined2014 — APPROVED** [review]

File: infoschema/tables.go [hunk 1]
```diff
@@ -1486,6 +1489,17 @@ var tablePlacementPoliciesCols = []columnInfo{
 	{name: "LEARNERS", tp: mysql.TypeLonglong, size: 64},
 }
 
+var tableVariablesInfoCols = []columnInfo{
+	{name: "VARIABLE_NAME", tp: mysql.TypeVarchar, size: 64, flag: mysql.NotNullFlag},
+	{name: "VARIABLES_SCOPE", tp: mysql.TypeVarchar, size: 64, flag: mysql.NotNullFlag},
```
Comment (tangenta) [inline infoschema/tables.go:1494]: `VARIABLES_SCOPE` -> `VARIABLE_SCOPE`

Is it a typo?

Comment (morgo) [reply infoschema/tables.go:1494]: Yeah, I agree it should be variable scope.

**Pull request merged**